### PR TITLE
feat(FCM): support Firebase Cloud Messaging plugin

### PR DESCRIPTION
**Pull request by klirix**
Added Firebase Cloud Messaging plugin.

Tested it out on Android, test on iOS is coming.

Event functions (onTokenChange, onNotification) repurposed to be observables.

Code feedback would be great, as it is my first plugin wrapper.

### CODE REVIEW

File: src/@ionic-native/plugins/fcm/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+/**
+ * @name FCM
+ * @description
+ * Provides basic functionality for Firebase Cloud Messaging
+ *
+ * @usage
+ * ```
+ * import { FCM } from 'ionic-native';
```
Comment (MTechDE) [inline src/@ionic-native/plugins/fcm/index.ts]: Please add typescript to the markup ```typescript

File: src/@ionic-native/plugins/fcm/index.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * constructor(private fcm: FCM) {
```
Comment (MTechDE) [inline src/@ionic-native/plugins/fcm/index.ts]: The usage part is a bit to short. Can you describe the plugin a bit more? Orientate you by other plugins.
Don't put the plugin method call in to the constructor.

Comment (klirix) [inline src/@ionic-native/plugins/fcm/index.ts]: Added additional examples in the **usage** section and in docs for each function

File: src/@ionic-native/plugins/fcm/index.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+   * 
+   * tokenChange.unsubscribe()
+   * 
+   * ```
+   * 
+   * @returns {Observable<string>} Returns an Observable that notifies with the change of device's registration id
+   */
+  @Cordova({
+    observable: true,
+    successIndex: 0
```
Comment (ihadeed) [inline src/@ionic-native/plugins/fcm/index.ts]: this line is not needed

File: src/@ionic-native/plugins/fcm/index.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  repo: 'https://github.com/fechanique/cordova-plugin-fcm',
+  platforms: ['iOS', 'Android']
+})
+@Injectable()
+export class FCM extends IonicNativePlugin {
+
+  /**
+   * Get's device's current registration id
+   * 
+   * ```typescript
```
Comment (ihadeed) [inline src/@ionic-native/plugins/fcm/index.ts]: Keep all usage examples in the main block above the `@Plugin` decorator. Do not add any usage documentation above the method declaration.

File: src/@ionic-native/plugins/fcm/index.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+   * fcm.subscribeToTopic('marketing');
+   * 
+   * ```
+   * 
+   * @param {string} topic Topic to be subscribed to
+   * 
+   * @returns {Promise<any>} Returns a promise resolving in result of subscribing to a topic
+   */
+  @Cordova({
+    callbackOrder: 'reverse'
```
Comment (ihadeed) [inline src/@ionic-native/plugins/fcm/index.ts]: This method should not have `callbackOrder: 'reverse'`. See https://github.com/fechanique/cordova-plugin-fcm/blob/master/www/FCMPlugin.js#L8

File: src/@ionic-native/plugins/fcm/index.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+   * fcm.unsubscribeToTopic('marketing');
+   * 
+   * ```
+   * 
+   * @param {string} topic Topic to be unsubscribed from
+   * 
+   * @returns {Promise<any>} Returns a promise resolving in result of unsubscribing from a topic
+   */
+  @Cordova({
+    callbackOrder: 'reverse'
```
Comment (ihadeed) [inline src/@ionic-native/plugins/fcm/index.ts]: This method should not have `callbackOrder: 'reverse'`. See https://github.com/fechanique/cordova-plugin-fcm/blob/master/www/FCMPlugin.js#L12

**Pull request merged**